src/abstract_trees.jl

```diff
@@ -0,0 +1,131 @@
+"""
+Implementation of the `AbstractTrees.jl`-interface 
+(see: [AbstractTrees.jl](https://github.com/JuliaCollections/AbstractTrees.jl)).
+
+The functions `children` and `printnode` make up the interface traits of `AbstractTrees.jl` 
+(see below for details).
+
+The goal of this implementation is to wrap a `DecisionTree` in this abstract layer, 
+so that a plot recipe for visualization of the tree can be created that doesn't rely 
+on any implementation details of `DecisionTree`. That opens the possibility to create
+a plot recipe which can be used by a variety of tree-like models. Actually there is 
+the idea to extend `MLJ.jl` in the future with a collection of such (generic) recipes.
+
+For a more detailed explanation of this concept have a look at the follwing article 
+in "Towards Data Science": 
+["If things are not ready to use"](https://towardsdatascience.com/part-iii-if-things-are-not-ready-to-use-59d2db378bec)
+"""
+
+
+"""
+    InfoNode{S, T}
+    InfoLeaf{T}
+
+These types are introduced so that additional information currently not present in 
+a `DecisionTree`-structure -- namely the feature names and the class labels -- 
+can be used for visualization. This additional information is stored in the attribute `info` of 
+these types. It is a `NamedTuple`. So it can be used to store arbitraty information, 
+apart from the two points mentioned.
+
+In analogy to the type definitions of `DecisionTree`, the generic type `S` is 
+the type of the feature values used within a node as a threshold for the splits
+between its children and `T` is the type of the classes given (these might be ids or labels).
+"""
+struct InfoNode{S, T}
+    node    :: DecisionTree.Node{S, T}
+    info    :: NamedTuple
+end
+
+struct InfoLeaf{T}
+	leaf    :: DecisionTree.Leaf{T}
+    info    :: NamedTuple
```

Comment:
There was a tab character that was leading to different indentation, at least in this GH display. I don't think you intended it. Here's a fix. 

```suggestion
    leaf    :: DecisionTree.Leaf{T}
    info    :: NamedTuple
```

Reply:
Ah ok, this didn't show up in my editors. I've replaced now all whitespace characters within these lines with tabs or spaces (hopefully in a consistent way).